system/lib/libc/emscripten_memcpy_bulkmem.S

```diff
@@ -7,10 +7,16 @@
 .globl _emscripten_memcpy_bulkmem
 _emscripten_memcpy_bulkmem:
   .functype _emscripten_memcpy_bulkmem (PTR, PTR, PTR) -> (PTR)
-  local.get 0
-  local.get 1
   local.get 2
-  memory.copy 0, 0
+#ifdef __wasm64__
+  i32.wrap_i64
+#endif
+  if
+    local.get 0
+    local.get 1
+    local.get 2
+    memory.copy 0, 0
+  end_if
```

Comment:
Why move the comparison here?   Maybe better to leave it (duplicated) in the C code where it is easier to read and debug.

Also, `_emscripten_memcpy_bulkmem` would then be replaced by the llvm intrinsic for the memory.copy instruction if we keep it as simple as it is today.

Reply:
I believe there actually is not a memory.copy intrinsic in LLVM currently. I did try a version of this with inline asm, and it did actually seem to work. But I wasn't sure whether we want to use inline asm.

If we do keep the version with the asm file, I think I prefer having the branch in the asm. I'm not sure whether it's any smaller, but I like that the asm function actually correctly implements memcpy (especially if we are going to keep calling the file and function "memcpy_bulkmem"). I think that's less error-prone, since if another use were to be introduced, we could have this bug all over again. This PR already fixes the same bug in memset too.

Reply:
I guess I was hoping that one day llvm might add a `__wasm_memory_copy` intrinsic and we could drop the asm version.

I'm fine with renaming the asm version we have to `__wasm_memory_copy` so it more obvious what it does.   I'm also fine with leaving the check in the asm file, but maybe we an explanatory comment?

Reply:
I don't mind adding builtins for memory copy and fill. I'd rather do that as a followup to the feature change though. I'm fine adding a comment to the asm files here, or just using inline asm for now and dropping the asm files.